tests/testsuite/git.rs

```diff
@@ -380,6 +382,109 @@ hello world
         .run();
 }
 
+#[cargo_test]
+fn dependency_in_submodule_via_path_base() {
+    // Using a submodule prevents the dependency from being discovered during the directory walk,
+    // so Cargo will need to follow the path dependency to discover it.
+
+    let git_project = git::new("dep1", |project| {
+        project
+            .file(
+                "Cargo.toml",
+                r#"
+                    cargo-features = ["path-bases"]
+
+                    [package]
+                    name = "dep1"
+                    version = "0.5.0"
+                    edition = "2015"
+                    authors = ["carlhuda@example.com"]
+
+                    [dependencies]
+                    dep2 = { version = "0.5.0", base = "submodules", path = "dep2" }
+
+                    [lib]
+                    name = "dep1"
+                "#,
+            )
+            .file(
+                "src/dep1.rs",
+                r#"
+                    extern crate dep2;
+
+                    pub fn hello() -> &'static str {
+                        dep2::hello()
+                    }
+                "#,
+            )
+    });
+
+    let git_project2 = git::new("dep2", |project| {
+        project
+            .file("Cargo.toml", &basic_lib_manifest("dep2"))
+            .file(
+                "src/dep2.rs",
+                r#"
+                    pub fn hello() -> &'static str {
+                        "hello world"
+                    }
+                "#,
+            )
+    });
+
+    let repo = git2::Repository::open(&git_project.root()).unwrap();
+    let url = git_project2.root().to_url().to_string();
+    git::add_submodule(&repo, &url, Path::new("submods/dep2"));
+    git::commit(&repo);
+
+    write_config_at(
+        paths::home().join(".cargo/config.toml"),
+        r#"
+            [path-bases]
+            submodules = '../dep1/submods'
+        "#,
+    );
```

Comment:
imo writing path-bases on behalf of git dependencies seems very questionable
- Constructing the right absolute path is a mess
- Constructing the right relative path is a mess

If you want to test `RecursivePathSource`, https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#paths-overrides is another user of that tyoe